src/python/pants/backend/typescript/dependency_inference/rules.py

```diff
@@ -128,7 +127,8 @@ async def infer_typescript_source_dependencies(
                 _determine_import_from_candidates(
                     candidates,
                     candidate_pkgs,
-                    file_extensions=TS_FILE_EXTENSIONS + JS_FILE_EXTENSIONS,
+                    tsconfig=maybe_config.ts_config,
+                    file_extensions=TS_FILE_EXTENSIONS + JS_FILE_EXTENSIONS + TSX_FILE_EXTENSIONS,
```

Comment:
If we're supporting TSX here, should we also support JSX? I'm not sure, just pattern matching.